src/Model/EndpointLocator.php

```diff
@@ -0,0 +1,209 @@
+    {
+        return $this->_instances[$alias] = $object;
+    }
+
+    /**
+     * Get a endpoint instance from the registry.
+     *
+     * @param string $alias The alias name you want to get.
+     * @param array $options The options you want to build the endpoint with.
+     * @return \Muffin\Webservice\Model\Endpoint
+     * @throws \RuntimeException If the registry alias is already in use.
+     */
+    public function get($alias, array $options = [])
+    {
+        if (isset($this->_instances[$alias])) {
+            if (!empty($options) && $this->_options[$alias] !== $options) {
+                throw new RuntimeException(sprintf(
+                    'You cannot configure "%s", it already exists in the registry.',
+                    $alias
+                ));
+            }
+
+            return $this->_instances[$alias];
+        }
+
+        list(, $classAlias) = pluginSplit($alias);
+        $options = ['alias' => $classAlias] + $options;
+
+        if (empty($options['className'])) {
+            $options['className'] = Inflector::camelize($alias);
+        }
+        $className = App::className($options['className'], 'Model/Endpoint', 'Endpoint');
+        if ($className) {
+            $options['className'] = $className;
+        } else {
+            if (!isset($options['endpoint']) && strpos($options['className'], '\\') === false) {
+                list(, $endpoint) = pluginSplit($options['className']);
+                $options['endpoint'] = Inflector::underscore($endpoint);
+            }
+            $options['className'] = 'Muffin\Webservice\Model\Endpoint';
+        }
+
+        if (empty($options['connection'])) {
+            if ($options['className'] !== 'Muffin\Webservice\Model\Endpoint') {
+                $connectionName = $options['className']::defaultConnectionName();
+            } else {
+                $pluginParts = explode('/', pluginSplit($alias)[0]);
+
+                $connectionName = Inflector::underscore(end($pluginParts));
+            }
+
+            $options['connection'] = ConnectionManager::get($connectionName);
+        }
+
+        $options['registryAlias'] = $alias;
+        $this->_instances[$alias] = $this->_create($options);
+        $this->_options[$alias] = $options;
+
+        return $this->_instances[$alias];
+    }
+
+    /**
+     * Check to see if an instance exists in the registry.
+     *
+     * @param string $alias The alias to check for.
+     * @return bool
+     */
+    public function exists($alias)
+    {
+        return isset($this->_instances[$alias]);
+    }
+
+    /**
+     * Returns configuration for an alias or the full configuration array for all aliases.
+     *
+     * @param string|null $alias Endpoint alias
+     * @return array
+     */
+    public function getConfig($alias = null)
+    {
+        if ($alias === null) {
+            return $this->_config;
+        }
+
+        return isset($this->_config[$alias]) ? $this->_config[$alias] : [];
```

Comment:
Does `TableLocator` return empty array or `null` if alias does not exist? Generally it should be `null`.

Reply:
[This is a copy and paste from TableLocator](https://github.com/cakephp/cakephp/blob/master/src/ORM/Locator/TableLocator.php#L88-L101). I'm unsure of the return type, I think empty array would be more consistent, as it means the data type return will always be the same. Which will help with moving to PHP 7.x return type hints.